### ISSUE TITLE
``BlockingIOError`` on `git commit` on Mac OSX

### ISSUE THREAD

**Issue by dotsdl** [xonsh/xonsh#2579]
When doing a `git commit` on Mac OS X under xonsh, I get:

```
Exception in thread Thread-34:
Traceback (most recent call last):
  File "/usr/local/miniconda3/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/usr/local/miniconda3/lib/python3.6/site-packages/xonsh/__amalgam__.py", line 11270, in run
    i = self._read_write(procout, stdout, sys.__stdout__)
  File "/usr/local/miniconda3/lib/python3.6/site-packages/xonsh/__amalgam__.py", line 11313, in _read_write
    stdbuf.flush()
BlockingIOError: [Errno 35] write could not complete without blocking
```

I'm using `xonsh` in a miniconda installation. My `xonfig` output looks like this, for reference:

```
+------------------+----------------------+
| xonsh            | 0.6.0                |
| Git SHA          | c29819c5             |
| Commit Date      | Nov 22 17:51:52 2017 |
| Python           | 3.6.3                |
| PLY              | 3.9                  |
| have readline    | True                 |
| prompt toolkit   | 1.0.15               |
| shell type       | prompt_toolkit       |
| pygments         | 2.2.0                |
| on posix         | True                 |
| on linux         | False                |
| on darwin        | True                 |
| on windows       | False                |
| on cygwin        | False                |
| is superuser     | False                |
| default encoding | utf-8                |
| xonsh encoding   | utf-8                |
| encoding errors  | surrogateescape      |
+------------------+----------------------+
```

Any ideas on how to proceed? Is anyone else experiencing this problem?

**Comment by dotsdl** [comment]
Some more information: this happens if I'm using neovim as my git editor, but not for vim. Using vim instead is a decent workaround, but I'm definitely interested in the root cause.

**Comment by gforsyth** [comment]
Hey @dotsdl -- sorry for the delay in getting to this! Can you try adding this to your `.xonshrc` and see if the problem persists?

``` 
__xonsh_commands_cache__.threadable_predictors['neovim'] = lambda args: False
```

**Comment by gforsyth** [comment]
Specific workaround to close this out is to add `neovim` to [threadable_predictors](https://github.com/xonsh/xonsh/blob/master/xonsh/commands_cache.py#L388-L437) with a `predict_false`.

**Comment by gforsyth** [comment]
Ideally we'd figure out a couple of ways to more generally predict / figure out if a command is going to use something like `ncurses` -- we have at least one of those somewhere in the code base.
But for now, yes, when commands break we can put them in threadable predictors.

**Comment by scopatz** [comment]
> Ideally we'd figure out a couple of ways to more generally predict / figure out if a command is going to use something like ncurses

Yeah, for reference, this code lives in the CommandsCache

**Comment by gforsyth** [comment]
Fixed a while back in #2240
